File: apps/backend/src/main/java/com/digisphere/pontofacil/domain/adress/Address.java
package com.digisphere.pontofacil.domain.adress;

import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Embeddable
@AllArgsConstructor
@NoArgsConstructor
public class Address {

  private String street;
  private Integer number;
  private String city;
  private String neighborhood;

  @Enumerated(EnumType.STRING)
  private UF UF;

}

File: apps/backend/src/main/kotlin/com/digisphere/pontofacil/company/domain/Address.kt
package com.digisphere.pontofacil.company.domain

import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class Address(
  val street: String,
  val number: Int,
  val city: String,
  val neighborhood: String,
  @Enumerated(EnumType.STRING)
  val UF: UF
)
